### ISSUE TITLE
Logging detected object to CSV from whole image in a folder

### ISSUE THREAD

**Issue by singgihbekti** [ultralytics/yolov5#8259]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello everyone, 
I am new to coding and now work on a detection object in some classes, let's say 5 classes.  I am trying to make a logging result to CSV from the detected object in a folder. The issue is how to make a summary of 5 classes from a target detected folder. The CSV or Summary in TXT is generated to help in showing the detailed detected object and summarize each counted object.

I try to make a logging result to CSV. 
I have tried to edit the ``` detect.py ``` but I haven't got the proper result. 
Previously, I use ``` --save-txt -- ``` to produce new labels from unlabelled images. 
``` python detect.py --source=C:\Users\.....\.......\target_folder --weights=C:\Users\.....\.......\project_folder\yolov5-master\runs\train\yolov5x6_results\weights\best.pt --save-txt --conf=0.70 ``` 

Instead, we need to make some changes in ```detect.py``` for CSV format, can we apply ``` --save-txt -- ``` for this purpose?

Does yolov5 provide this kind of logging or what part should be edited correctly for this purpose?
Thank you

**Comment by glenn-jocher** [comment]
@singgihbekti 👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) models allow for simple model loading and inference in a pure python environment without using `detect.py`. You can export detections to pandas formats and then onward to any other format you need like JSON or CSV.

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://docs.ultralytics.com/yolov5/tutorials/model_export) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
#                                            'custom', 'path/to/best.pt')  # custom model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.

results.xyxy[0]  # im predictions (tensor)
results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) for details.

Good luck 🍀 and let us know if you have any other questions!

**Comment by singgihbekti** [comment]
hi @glenn-jocher , thank you for answer. It is so clear answer for running online in Google Colab. 
In case I run my inference offline, 
I have tried:
1. Can we edit the ```detect.py``` ? which part should be edited?

I have added ```import pandas``` 
and I call ```results.pandas().xyxy[0]``` at # Write results in line 163

but It doesn't work.


2. I make new file ```.py``` with the code below to execute it, but got some errors and doesn't work

````
import cv2
import torch
import pandas

# Model
model = 'C:/Users/ESID/Downloads/SB-TRIAL/yolov5/yolov5-40-my3plus/yolov5-master/runs/train/yolov5l_results/weights/best.pt'
# Images
im = 'C:/Users/ESID/Downloads/SB-TRIAL/yolov5/yolov5-40-my3plus/1a.2._Labelled_Img_Split/valid/images'

# Inference
results = model(im)
results.pandas().xyxy[0]

````

how to solve for both ways?
thank you

**Comment by glenn-jocher** [comment]
@singgihbekti Thank you for your questions! When running YOLOv5 for offline inference, you may encounter an error when attempting to modify the `detect.py` script or creating a new script. 
To handle this, you can modify the `detect.py` script as described below:

1. Open `detect.py` and navigate to line 163.
2. Add the code `results.save('results.csv')` to save the inference results to a CSV file.

For your second case, when creating a new Python script:

1. Modify your code to load the model using `model = torch.hub.load('ultralytics/yolov5', 'custom', path_or_model='path/to/best.pt')`.
2. Then provide the image path as a string, not as a directory using `im = 'C:/path/to/image.jpg'`.
3. Lastly, call `results.save('results.csv')` to save the inference results to a CSV file.

By following these steps, you should be able to save the inference results to a CSV file. Let me know if you encounter any further issues!
